src/reps/rep-utils.js

```diff
@@ -157,6 +160,66 @@ function wrapRender(renderMethod) {
   };
 }
 
+function isGripSelectableInInspector(grip) {
+  return grip
+    && typeof grip === "object"
+    && grip.preview
+    && [
+      nodeConstants.TEXT_NODE,
+      nodeConstants.ELEMENT_NODE
+    ].includes(grip.preview.nodeType);
+}
+
+function getFlattenedGrips(parameters, filterFn) {
+  if (!Array.isArray(parameters)) {
+    return null;
+  }
+  let nodeGrips = parameters.reduce((result, parameter) => {
+    return result.concat(getPreviewGrips(parameter, filterFn));
+  }, []);
+  return [...new Set(nodeGrips)];
+}
+
+function getPreviewGrips(grip, filterFn = null) {
+  let grips = [];
+  if (!filterFn || filterFn(grip)) {
+    grips.push(grip);
+  }
+
+  let previewItems = [];
+  if (grip && grip.preview) {
+    if (grip.preview.items) {
```

Comment:
Some comments would be nice in this section, to know which block corresponds to which kind of grip.